### PR TITLE
ITSO Parser (UK)

### PR DESCRIPTION
**Pull request by gsp8181**
# What's new

- ITSO parser added to NFC app. ITSO is a standard for UK transport cards (aside from Oyster). Used for rail, bus and subway passes across GB. This parser grabs the card number and expiry date. More information is available on the card but locked behind propriety apps and an extensive database of station stops and operators is needed.

# Verification 

- Scan an ITSO card and verify that the number matches the number printed on the card. Expiry date will match if printed on the card.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by gornekich** [comment]
Hello @gsp8181 ! Thanks for PR!
Could you please share example .nfc file to verify your PR? You can send it here or DM me in Discord gornekich

File: applications/main/nfc/plugins/supported_cards/itso.c [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            FURI_HAL_RTC_IS_LEAP_YEAR(datetime->year), datetime->month);
+        (datetime->month)++;
+    }
+
+    datetime->day = days + 1;
+    datetime->hour = seconds_in_day / FURI_HAL_RTC_SECONDS_PER_HOUR;
+    datetime->minute =
+        (seconds_in_day % FURI_HAL_RTC_SECONDS_PER_HOUR) / FURI_HAL_RTC_SECONDS_PER_MINUTE;
+    datetime->second = seconds_in_day % FURI_HAL_RTC_SECONDS_PER_MINUTE;
+}
```
Comment (gornekich) [inline applications/main/nfc/plugins/supported_cards/itso.c]: Please, move this function to furi_hal_rtc.h and rename to furi_hal_rtc_timestamp_to_datetime().

File: applications/main/nfc/plugins/supported_cards/itso.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+#include <applications/services/locale/locale.h>
+#include <furi_hal_rtc.h>
+
+#define FURI_HAL_RTC_SECONDS_PER_MINUTE 60
+#define FURI_HAL_RTC_SECONDS_PER_HOUR (FURI_HAL_RTC_SECONDS_PER_MINUTE * 60)
+#define FURI_HAL_RTC_SECONDS_PER_DAY (FURI_HAL_RTC_SECONDS_PER_HOUR * 24)
+#define FURI_HAL_RTC_EPOCH_START_YEAR 1970
+#define FURI_HAL_RTC_IS_LEAP_YEAR(year) \
+    ((((year) % 4 == 0) && ((year) % 100 != 0)) || ((year) % 400 == 0))
```
Comment (gornekich) [inline applications/main/nfc/plugins/supported_cards/itso.c]: After moving timestamp_to_datetime() function to furi_hal_rtc delete this defines here

**Review by gornekich — CHANGES_REQUESTED** [review]
Hello @gsp8181 
Sorry for late review. I tested you PR, everything works great!
Please, make some changes to make code look better and we will merge PR.

**Comment by gornekich** [comment]
By the way, we already have #3294 which introduces furi_hal_rtc_timestamp_to_datetime() function. Let's wait when that PR is merged and rework this PR with this function

**Comment by skotopes** [comment]
Un-draft when ready

**Review by hedger — APPROVED** [review]

**Review by gornekich — APPROVED** [review]

**Pull request merged**